src/fields/fft_poisson_solver/fft/WrapCuDST.cpp

```diff
@@ -67,81 +70,316 @@ namespace AnyDST
+                amrex::Gpu::SharedMemory<amrex::Real> gsm;
+                amrex::Real* const tile = gsm.dataPtr();
+
+                const int thread_x = threadIdx.x&(tile_dim-1);
+                const int thread_y = threadIdx.x>>5; // for tile_dim = 2^5
+                const int block_y = blockIdx.x/num_blocks_x;
+                const int block_x = blockIdx.x - block_y*num_blocks_x;
+                int mat_x = block_x * tile_dim + thread_x;
+                int mat_y = block_y * tile_dim + thread_y;
+
+                for (int i = 0; i < tile_dim; i += block_rows) {
+                    if(mat_x < n_data && (mat_y+i) < n_batch) {
+                        tile[(thread_y+i)*(tile_dim+1) + thread_x] = in[(mat_y+i)*n_data + mat_x];
+                    }
+                }
+
+                __syncthreads();
+
+                mat_x = block_y * tile_dim + thread_x;
+                mat_y = block_x * tile_dim + thread_y;
+
+                for (int i = 0; i < tile_dim; i += block_rows) {
+                    if(mat_x < n_batch && (mat_y+i) < n_data) {
+                        out[(mat_y+i)*n_batch + mat_x] = tile[thread_x*(tile_dim+1) + thread_y+i];
+                    }
+                }
+            });
+    };
+
     DSTplan CreatePlan (const amrex::IntVect& real_size, amrex::FArrayBox* position_array,
                         amrex::FArrayBox* fourier_array)
     {
         HIPACE_PROFILE("AnyDST::CreatePlan()");
         DSTplan dst_plan;
-        const int nx = real_size[0];
-        const int ny = real_size[1];
-
-        // Allocate expanded_position_array Real of size (2*nx+2, 2*ny+2)
-        // Allocate expanded_fourier_array Complex of size (nx+2, 2*ny+2)
-        amrex::Box expanded_position_box {{0, 0, 0}, {2*nx+1, 2*ny+1, 0}};
-        amrex::Box expanded_fourier_box {{0, 0, 0}, {nx+1, 2*ny+1, 0}};
-        dst_plan.m_expanded_position_array =
-            std::make_unique<amrex::FArrayBox>(
-                expanded_position_box, 1);
-        dst_plan.m_expanded_fourier_array =
-            std::make_unique<amrex::BaseFab<amrex::GpuComplex<amrex::Real>>>(
-                expanded_fourier_box, 1);
-
-        // setting the initial values to 0
-        // we don't set the expanded Fourier array, because it will be initialized by the FFT
-        dst_plan.m_expanded_position_array->setVal<amrex::RunOn::Device>(0.,
-            dst_plan.m_expanded_position_array->box(), 0,
-            dst_plan.m_expanded_position_array->nComp());
-
-        const amrex::IntVect& expanded_size = expanded_position_box.length();
-
-        // Initialize fft_plan.m_plan with the vendor fft plan.
-        cufftResult result;
-        result = cufftPlan2d(
-            &(dst_plan.m_plan), expanded_size[1], expanded_size[0], VendorR2C);
 
-        if ( result != CUFFT_SUCCESS ) {
-            amrex::Print() << " cufftplan failed! Error: " <<
-                CuFFTUtils::cufftErrorToString(result) << "\n";
+        amrex::ParmParse pp("fields");
```

Comment:
```suggestion
        amrex::ParmParse pp("hipace");
```

I would prefer to parse for `"hipace"`.  We do this for other field related things like the `bxby_solver` as well and this would be the only parsing for `fields`. Maybe we can separate these at some point, but for convenience I would stick with `hipace` for now.